packages/apollo-gateway/src/__tests__/gateway/lifecycle-hooks.test.ts

```diff
@@ -0,0 +1,153 @@
+import { ApolloGateway, GatewayConfig } from '../../index';
+import * as accounts from '../__fixtures__/schemas/accounts';
+import * as books from '../__fixtures__/schemas/books';
+import * as inventory from '../__fixtures__/schemas/inventory';
+import * as product from '../__fixtures__/schemas/product';
+import * as reviews from '../__fixtures__/schemas/reviews';
+
+const services = [product, reviews, inventory, accounts, books];
+const serviceDefinitions = services.map((s, i) => ({
+  name: s.name,
+  typeDefs: s.typeDefs,
+  url: `http://localhost:${i}`,
+}));
+
+describe('lifecycle hooks', () => {
+  it('uses updateServiceDefinitions override', async () => {
+    const experimental_updateServiceDefinitions = jest.fn(
+      async (config: GatewayConfig) => {
+        return [serviceDefinitions, true];
+      },
+    );
+
+    const gateway = new ApolloGateway({
+      serviceList: serviceDefinitions,
+      experimental_updateServiceDefinitions,
+      experimental_didUpdateComputedFederationConfig: jest.fn(),
+    });
+
+    await gateway.load();
+
+    expect(experimental_updateServiceDefinitions).toBeCalled();
```

Comment:
nit: should we also validate that its return value is now a property of the gateway's state? Since this has a side effect that can propagate to the state, it seems like good practice to check for a side effect taking place, unlike the other observability functions, which we can just check if they're called.